distributed/scheduler.py

```diff
@@ -51,6 +53,9 @@
 DEFAULT_DATA_SIZE = config.get('default-data-size', 1000)
 
 
+# XXX avoid inheriting from Server? there is some large potential for confusion
+# between base and derived attribute namespaces...
```

Comment:
Scheduler and other heavy objects (objects with many of their own attributes and methods) inherit from `Server`, which makes it more confusing when looking up attributes (does it stem from Scheduler or Server?) and also makes it possible to shadow some Server attributes with some Scheduler attributes by mistake (that may actually have been the case with `address` or `port`). Using composition would solve that issue.